sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -341,6 +341,62 @@ class SQLMetricsSuite extends SharedSparkSession with SQLMetricsTestUtils
     }
   }
 
+  test("ShuffledHashJoin(outer) metrics") {
+    withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "40",
+      SQLConf.SHUFFLE_PARTITIONS.key -> "2",
+      SQLConf.PREFER_SORTMERGEJOIN.key -> "false") {
+      val df1 = Seq((1, "1"), (2, "2")).toDF("key", "value")
+      val df2 = (1 to 10).map(i => (i, i.toString)).toSeq.toDF("key2", "value")
+
+      Seq(("right_outer", 0L, df1, df2, false), ("left_outer", 0L, df2, df1, false),
+        ("right_outer", 0L, df1, df2, true), ("left_outer", 0L, df2, df1, true))
+        .foreach { case (joinType, nodeId, df1, df2, enableWholeStage) =>
```

Comment:
nit: `df1` -> `leftDf` and `df2` -> `rightDf`

Reply:
fixed in the latest commit